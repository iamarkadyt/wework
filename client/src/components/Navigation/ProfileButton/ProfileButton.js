import React from 'react'
import { Link } from 'react-router-dom'

const profileButton = ({
  isAuthenticated,
  toggleDropdown
}) => (
  isAuthenticated
    ? <button className="Navigation-button" onClick={toggleDropdown}>
        <span>Menu</span>
      </button> 
    : <Link to="/login">
        <span>Log In</span>
      </Link>
)

export default profileButton
