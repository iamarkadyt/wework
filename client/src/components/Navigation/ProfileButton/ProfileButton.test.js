import React from 'react'
import { shallow } from 'enzyme'
import ProfileButton from './ProfileButton'
import cloneDeep from 'lodash.clonedeep'

let mountedComponent, props

const getMockProps = () => {
  return cloneDeep({
    isAuthenticated: true
  })
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<ProfileButton {...props} />)
  }
  return mountedComponent
}

describe('ProfileButton', () => {
  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  describe('props', () => {
    describe('effects', () => {
      describe('isAuthenticated', () => {
        it('if true, Account button is rendered', () => {
          props.isAuthenticated = true
          expect(comp().find('button[className="Navigation-button"]').exists()).toBe(true)
        })

        it('if false, Link component is rendered', () => {
          props.isAuthenticated = false
          expect(comp().find('Link').exists()).toBe(true)
        })
      })
    })
  })

  it('matches snapshot with isAuthenticated set to true', () => {
    props.isAuthenticated = true
    expect(comp()).toMatchSnapshot()
  })

  it('matches snapshot with isAuthenticated set to false', () => {
    props.isAuthenticated = false
    expect(comp()).toMatchSnapshot()
  })
})

