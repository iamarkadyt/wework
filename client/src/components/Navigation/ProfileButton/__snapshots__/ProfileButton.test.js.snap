// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileButton matches snapshot with isAuthenticated set to false 1`] = `
<Link
  replace={false}
  to="/login"
>
  <span>
    Log In
  </span>
</Link>
`;

exports[`ProfileButton matches snapshot with isAuthenticated set to true 1`] = `
<button
  className="Navigation-button"
  onClick={[MockFunction]}
>
  <span>
    Account
  </span>
</button>
`;
